### PR TITLE
API users can turn off key#7071

### PR DESCRIPTION
**Pull request by shabeeb-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/7071

*Description of changes:*
* Added new page for developer access

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/public/config/configuration.json [hunk 1]
```diff
@@ -25,5 +25,6 @@
   "LEGAL_API_VERSION": "/api/v1",
   "ONE_STOP_URL": "https://dev.onestop.gov.bc.ca/",
   "CORPORATE_ONLINE_URL": "https://dev.corponline.gov.bc.ca/",
-  "FAS_WEB_URL": "https://fas-dev.apps.silver.devops.gov.bc.ca/"
+  "FAS_WEB_URL": "https://fas-dev.apps.silver.devops.gov.bc.ca/",
+  "API_DOCUMENTATION_URL": "https://fas-dev.apps.silver.devops.gov.bc.ca/"
```
Comment (shabeeb-aot) [inline auth-web/public/config/configuration.json:29]: need to update in next PR

File: auth-web/package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
   },
   "dependencies": {
     "@bcrs-shared-components/corp-type-module": "1.0.4",
-    "@mdi/font": "^4.5.95",
+    "@mdi/font": "^6.4.95",
```
Comment (shabeeb-aot) [inline auth-web/package.json:17]: updated mdi fort to support new icons

File: auth-web/src/components/auth/common/AccountLoginOptionPicker.vue [hunk 3]
```diff
@@ -107,7 +107,7 @@ export default class AccountLoginOptionPicker extends Mixins(AccountChangeMixin,
       type: LoginSource.BCSC,
       title: this.$t('bCSCLoginOptionTitle'),
       description: this.$t('bCSCLoginOptionDescription'),
-      icon: 'mdi-account-card-details-outline'
+      icon: 'mdi-card-account-details-outline'
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/common/AccountLoginOptionPicker.vue:110]: due to update of mdi icons

File: auth-web/src/views/auth/AccountSettings.vue [hunk 4]
```diff
@@ -197,6 +197,34 @@
...
+          <!-- add inside permission when adding menu items in this list -->
+            <v-list-item-group color="primary" role="list">
+              <v-subheader class="mt-2 px-0">ADVANCED SETTINGS</v-subheader>
+               <!-- v-can:MANAGE_STATEMENTS.hide -->
+              <v-list-item
+                dense
+                class="py-1 px-4"
+                aria-label="Developer Access"
+                role="listitem"
+                :to="developerAccessUrl"
```
Comment (shabeeb-aot) [inline auth-web/src/views/auth/AccountSettings.vue:215]: Need to add permission

File: auth-web/src/util/constants.ts [hunk 5]
```diff
@@ -216,6 +216,7 @@ export enum Permission {
     DEACTIVATE_ACCOUNT='DEACTIVATE_ACCOUNT',
     VIEW_USER_LOGINSOURCE='VIEW_USER_LOGINSOURCE',
     EDIT_BUSINESS_INFO = 'EDIT_BUSINESS_INFO',
+    VIEW_DEVELOPER_ACCESS = 'VIEW_DEVELOPER_ACCESS',
```
Comment (sumesh-aot) [inline auth-web/src/util/constants.ts:219]: Need to create a migration file to add this to ADMIN role

**Review by sumesh-aot — APPROVED** [review]

File: auth-web/src/util/constants.ts (hunk 5 above)
Comment (saravanpa-aot) [reply auth-web/src/util/constants.ts:219]: yeah..will do a PR now

File: auth-web/src/components/auth/account-settings/advance-settings/DeveloperAccess.vue [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import ConfigHelper from '@/util/config-helper'
+import ExistingAPIKeys from '@/components/auth/account-settings/advance-settings/ExistingAPIKeys.vue'
+
+@Component({
+  components: {
+    ExistingAPIKeys
+  }
+})
+export default class DeveloperAccess extends Mixins(AccountChangeMixin) {
+APIDOCUMENTATION_URL = ConfigHelper.apiDocumentationUrl() || ''
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/account-settings/advance-settings/DeveloperAccess.vue:36]: does it need an intendation?

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**